0024-pre-shared-key-v2/0024-pre-shared-key-v2.md

```diff
@@ -0,0 +1,287 @@
+---
+title: Pre-Shared Key V2 (PSKv2) Transport Protocol
+draft: 1
+---
+# Pre-Shared Key V2 (PSKv2) Transport Protocol
+
+This spec defines version 2 of the Pre-Shared Key Transport Protocol. PSKv2 implements end-to-end data encryption, condition and fulfillment generation, [End-to-End Quoting](#end-to-end-quoting) and [Chunked Payments](#chunked-payments). The approach to chunked payments is inspired by the Internet's Transmission Control Protocol (TCP).
+
+**Table of Contents**
+
+* [Overview](#overview)
+  + [Setup](#setup)
+  + [Encryption](#encryption)
+  + [Condition and Fulfillment Generation](#condition-and-fulfillment-generation)
+  + [End-To-End Quoting](#end-to-end-quoting)
+  + [Chunked Payments](#chunked-payments)
+* [Model of Operation](#model-of-operation)
+  + [Informational Quote](#informational-quote)
+  + [Single Payment](#single-payment)
+  + [Chunked Payment](#chunked-payment)
+* [Specification](#specification)
+  + [Encryption](#encryption-1)
+  + [Data Packet](#data-packet)
+* [Appendix A: Advanced Sender Features](#appendix-a-advanced-sender-features)
+
+## Overview
+
+Like [PSKv1](./0016-pre-shared-key/0016-pre-shared-key.md), PSKv2 uses a secret shared between the sender and receiver to generate the conditions and fulfillments for payments without repeated end-to-end communication. PSKv2 additionally includes support for End-to-End Quoting and Chunked Payments.
```

Comment:
What kinds of payments will PSK2 be used for? Will we use it for everything, or will micropayments (which we expect to be under the MTU) use a different protocol? A lot of the points in this document are very different depending on whether we're thinking about a large payment or a small payment. Most of the time, it seems like this spec is describing retail-sized payments that require lots of chunking.

If it's also being used for micropayments, I think that deserves more mentioning. For instance, many of the attacks described really don't work against micropayments. And many of the quoting mechanisms aren't needed when you're just sending a 1-chunk micropayment.